packages/amplify-graphql-index-transformer/src/resolvers/generators/rds-vtl-generator.ts

```diff
@@ -0,0 +1,80 @@
+import { TransformerContextProvider, TransformerResolverProvider } from '@aws-amplify/graphql-transformer-interfaces';
+import {
+  Expression,
+  printBlock,
+  compoundExpression,
+  set,
+  ref,
+  list,
+  qref,
+  methodCall,
+  str
+} from 'graphql-mapping-template';
+import { PrimaryKeyDirectiveConfiguration } from '../../types';
+import _ from 'lodash';
+import {
+  addIndexToResolverSlot,
+  getResolverObject,
+  mergeInputsAndDefaultsSnippet,
+  validateSortDirectionInput
+} from '../resolvers';
+import {
+  PrimaryKeyVTLGenerator,
+} from "./vtl-generator";
+
+export class RDSPrimaryKeyVTLGenerator implements PrimaryKeyVTLGenerator {
+  generate = (config: PrimaryKeyDirectiveConfiguration, ctx: TransformerContextProvider, resolverMap: Map<TransformerResolverProvider, string>): void => {
+    this.updateResolvers(config, ctx, resolverMap);
+  };
+
+  updateResolvers = (config: PrimaryKeyDirectiveConfiguration, ctx: TransformerContextProvider, resolverMap: Map<TransformerResolverProvider, string>): void => {
+    const getResolver = getResolverObject(config, ctx, 'get');
+    const listResolver = getResolverObject(config, ctx, 'list');
+    const createResolver = getResolverObject(config, ctx, 'create');
+    const updateResolver = getResolverObject(config, ctx, 'update');
+    const deleteResolver = getResolverObject(config, ctx, 'delete');
+
+    const primaryKeySnippet = this.setPrimaryKeySnippet(config);
+
+    if (getResolver) {
+      addIndexToResolverSlot(getResolver, [primaryKeySnippet]);
+    }
+
+    if (listResolver) {
+      const sortDirectionValidation = printBlock('Validate the sort direction input')(compoundExpression(validateSortDirectionInput(config, true)));
+      addIndexToResolverSlot(listResolver, [
+        primaryKeySnippet,
+        sortDirectionValidation
+      ]);
+    }
+
+    if (createResolver) {
+      addIndexToResolverSlot(createResolver, [primaryKeySnippet]);
+    }
+
+    if (updateResolver) {
+      addIndexToResolverSlot(updateResolver, [primaryKeySnippet]);
+    }
+
+    if (deleteResolver) {
+      addIndexToResolverSlot(deleteResolver, [primaryKeySnippet]);
+    }
+  };
```

Comment:
Do we need these for RDS?

Reply:
Yes we do, in order to set the `WHERE` conditions based on the columns used in primary key.